docs/dotnet/dotnet-aspnet-metrics.md

```diff
@@ -0,0 +1,196 @@
+# Semantic Conventions for ASP.NEt Core specific metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for ASP.NET Core metrics, not specific to HTTP
+
+**Disclaimer:** These are initial .NET metric instruments available in .NET 8 but more may be added in the future.
+
+<!-- toc -->
+
+- [Server](#server)
+- [Routing](#routing)
+  * [Metric: `aspnet.routing.match_attempts`](#metric-aspnetroutingmatch_attempts)
+- [Exception metrics](#exception-metrics)
+  * [Metric: `aspnet.diagnostics_handler.exceptions`](#metric-aspnetdiagnostics_handlerexceptions)
+- [Rate-limiting](#rate-limiting)
+  * [Metric: `aspnet.rate_limiting.active_request_leases`](#metric-aspnetrate_limitingactive_request_leases)
+  * [Metric: `aspnet.rate_limiting.request_lease.duration`](#metric-aspnetrate_limitingrequest_leaseduration)
+  * [Metric: `aspnet.rate_limiting.queued_requests`](#metric-aspnetrate_limitingqueued_requests)
+  * [Metric: `aspnet.rate_limiting.request.time_in_queue`](#metric-aspnetrate_limitingrequesttime_in_queue)
+  * [Metric: `aspnet.rate_limiting.requests`](#metric-aspnetrate_limitingrequests)
+
+<!-- tocstop -->
+
+## Server
+
+## Routing
+
+All routing metrics are reported by `Microsoft.AspNetCore.Routing` meter.
+
+### Metric: `aspnet.routing.match_attempts`
+
+<!-- semconv metric.aspnet.routing.match_attempts(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    |
+| -------- | --------------- | ----------- | -------------- |
+| `aspnet.routing.match_attempts` | Counter | `{match}` | Number of requests that were attempted to be matched to an endpoint. [1] |
+
+**[1]:** Meter name is `Microsoft.AspNetCore.Routing`.
+<!-- endsemconv -->
+
+<!-- semconv metric.aspnet.routing.match_attempts(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| `aspnet.routing.match_status` | string | Match result - success or failure | `success`; `failure` | Recommended |
+| `aspnet.routing.fallback` | string | TODO | `TODO` | Required |
+| `http.route` | string | The matched route (path template in the format used by the respective server framework). See note below [1] | `/users/:userID?`; `{controller}/{action}/{id?}` | Required |
```

Comment:
This is conditionally required if a route was matched

Reply:
should we drop `aspnet.routing.match_status` then and use lack of route to count unsuccessful matches?

Reply:
seems reasonable to me